src/language-css/parser-postcss.js

```diff
@@ -23,23 +31,32 @@ function parseValueNodes(nodes) {
 
   for (let i = 0; i < nodes.length; ++i) {
     const node = nodes[i];
-    const isUnquotedDataURLCall =
-      node.type === "func" &&
-      node.value === "url" &&
-      node.group &&
-      node.group.groups &&
-      node.group.groups[0] &&
-      node.group.groups[0].groups &&
-      node.group.groups[0].groups.length > 2 &&
-      node.group.groups[0].groups[0].type === "word" &&
-      node.group.groups[0].groups[0].value === "data" &&
-      node.group.groups[0].groups[1].type === "colon" &&
-      node.group.groups[0].groups[1].value === ":";
-
-    if (isUnquotedDataURLCall) {
-      node.group.groups = [stringifyGroup(node)];
-    }
 
+    if (node.type === "func" && node.value === "url") {
+      const groups = (node.group && node.group.groups) || [];
+
+      // Create a view with any top-level comma groups flattened.
+      let groupList = [];
+      for (let i = 0; i < groups.length; i++) {
+        const group = groups[i];
+        if (group.type === "comma_group") {
+          groupList = groupList.concat(group.groups);
+        } else {
+          groupList.push(group);
+        }
+      }
+
+      // Stringify if the value parser can't handle the content.
+      if (
+        hasSCSSInterpolation(groupList) ||
+        (!hasStringOrFunction(groupList) && !isSCSSVariable(groupList[0]))
+      ) {
+        const stringifiedContent = stringifyNode({
+          groups: node.group.groups,
+        });
+        node.group.groups = [stringifiedContent.trim()];
+      }
+    }
```

Comment:
Not sure it is a good solution, we really should fix it on parser side, it is still output broken url with interpolation and variables

Reply:
Broken in what way? Not formatted? Or really broken? Show an example please.